src/dlgMapper.cpp

```diff
@@ -75,39 +94,47 @@ dlgMapper::dlgMapper( QWidget * parent, Host * pH, TMap * pM )
+    connect(lineSize, qOverload<int>(&QSpinBox::valueChanged), this, &dlgMapper::slot_lineSize);
+    connect(roomSize, qOverload<int>(&QSpinBox::valueChanged), this, &dlgMapper::slot_roomSize);
+    connect(togglePanel, &QAbstractButton::pressed, this, &dlgMapper::slot_togglePanel);
+    connect(showArea, qOverload<const QString&>(&QComboBox::activated), mp2dMap, &T2DMap::slot_switchArea);
+    connect(dim2, &QAbstractButton::pressed, this, &dlgMapper::show2dView);
+    connect(showRoomIDs, &QCheckBox::stateChanged, this, &dlgMapper::slot_toggleShowRoomIDs);
 
+#if defined(INCLUDE_3DMAPPER)
+    connect(ortho, &QAbstractButton::pressed, glWidget, &GLWidget::fullView);
+    connect(singleLevel, &QAbstractButton::pressed, glWidget, &GLWidget::singleView);
+    connect(increaseTop, &QAbstractButton::pressed, glWidget, &GLWidget::increaseTop);
+    connect(increaseBottom, &QAbstractButton::pressed, glWidget, &GLWidget::increaseBottom);
+    connect(reduceTop, &QAbstractButton::pressed, glWidget, &GLWidget::reduceTop);
+    connect(reduceBottom, &QAbstractButton::pressed, glWidget, &GLWidget::reduceBottom);    
     connect(shiftZup, &QAbstractButton::pressed, glWidget, &GLWidget::shiftZup);
     connect(shiftZdown, &QAbstractButton::pressed, glWidget, &GLWidget::shiftZdown);
     connect(shiftLeft, &QAbstractButton::pressed, glWidget, &GLWidget::shiftLeft);
     connect(shiftRight, &QAbstractButton::pressed, glWidget, &GLWidget::shiftRight);
     connect(shiftUp, &QAbstractButton::pressed, glWidget, &GLWidget::shiftUp);
     connect(shiftDown, &QAbstractButton::pressed, glWidget, &GLWidget::shiftDown);
     connect(showInfo, &QAbstractButton::clicked, glWidget, &GLWidget::showInfo);
-    connect(showArea, qOverload<const QString&>(&QComboBox::activated), mp2dMap, &T2DMap::slot_switchArea);
     connect(defaultView, &QAbstractButton::pressed, glWidget, &GLWidget::defaultView);
-    connect(dim2, &QAbstractButton::pressed, this, &dlgMapper::show2dView);
     connect(sideView, &QAbstractButton::pressed, glWidget, &GLWidget::sideView);
     connect(topView, &QAbstractButton::pressed, glWidget, &GLWidget::topView);
-    connect(togglePanel, &QAbstractButton::pressed, this, &dlgMapper::slot_togglePanel);
-    connect(lineSize, qOverload<int>(&QSpinBox::valueChanged), this, &dlgMapper::slot_lineSize);
-    connect(roomSize, qOverload<int>(&QSpinBox::valueChanged), this, &dlgMapper::slot_roomSize);
     connect(scale, &QAbstractSlider::valueChanged, glWidget, &GLWidget::setScale);
     connect(xRot, &QAbstractSlider::valueChanged, glWidget, &GLWidget::setXRotation);
     connect(yRot, &QAbstractSlider::valueChanged, glWidget, &GLWidget::setYRotation);
     connect(zRot, &QAbstractSlider::valueChanged, glWidget, &GLWidget::setZRotation);
-    connect(showRoomIDs, &QCheckBox::stateChanged, this, &dlgMapper::slot_toggleShowRoomIDs);
+
+    glWidget->hide();
+#else
+    dim2->setDisabled(true);
```

Comment:
Why not `dim2->hide()` so that unsupported systems do not even suggest that there is a 3D map?

Reply:
So people aren't confused as to why their friend see the button and they don't :)